cthreads.h

```diff
@@ -90,6 +94,16 @@ struct cthreads_mutex {
   #endif
 };
 
+#ifdef CTHREADS_SEMAPHORE
```

Comment:
Any specific reasons to put it above mutex? If possible move it, so that it is the last one, like the definitions.